harvester_e2e_tests/scenarios/test_vm_actions.py

```diff
@@ -301,3 +301,38 @@ def test_update_vm_network(self, request, admin_session,
         assert found, 'Failed to add new network to VM %s' % (vm_name)
         # TODO(gyee): 1) make sure the new interface got an IP; and
         # 2) make sure we can ping that IP if it has a port in a public router
+
+    def test_validate_vol_inuse(self, request, admin_session,
+                                harvester_api_endpoints, basic_vm):
+        resp = admin_session.get(harvester_api_endpoints.get_vm_instance % (
+            basic_vm['metadata']['name']))
+        if resp.status_code == 200:
+            vm_data = resp.json()
+            vm_name = vm_data['metadata']['namespace'] + \
+                "/" + vm_data['metadata']['name']
+            vol_name = json.loads((
+                vm_data['metadata']['annotations'].get(
+                    'harvesterhci.io/diskNames')))
+            resp = admin_session.get(harvester_api_endpoints.get_volume % (
+                vol_name[0]))
+            vol_data = resp.json()
+            vol_vm_annotation = json.loads(
+                vol_data['metadata']['annotations'].get(
+                    'harvesterhci.io/owned-by'))
+            assert vol_vm_annotation[0].get('refs')[0] == vm_name
+
+    def test_delete_vol_inuse(self, request, admin_session,
+                              harvester_api_endpoints, basic_vm):
+        resp = admin_session.get(harvester_api_endpoints.get_vm_instance % (
```

Comment:
suggest using this instead https://github.com/harvester/tests/blob/main/harvester_e2e_tests/utils.py#L125 as it has error checking